addons/account/models/account_move_line.py

```diff
@@ -1811,6 +1800,17 @@ def get_accounting_rate(vals):
             else:
                 partial_credit_amount_currency = min_recon_amount
 
+        # Check if there is something left to reconcile. Move to the next loop iteration if not.
+        skip_reconciliation = False
+        if recon_currency.is_zero(partial_amount) and recon_currency.is_zero(recon_debit_amount):
+            res['debit_vals'] = None
+            skip_reconciliation = True
+        if recon_currency.is_zero(partial_amount) and recon_currency.is_zero(recon_credit_amount):
+            res['credit_vals'] = None
+            skip_reconciliation = True
+        if skip_reconciliation:
+            return res
```

Comment:
Hmmm that's interesting. I think you are right but I have to add tests about that and challenge all reconciliation cases. I'll check it next week.

Reply:
@ngochung207 Hello. I checked your PR and I don't understand what do you expect.

Currency company: VND, and rate USD: 23.000 VND/USD
Journal Item 1: with amount_currency: 100, currency_id: USD
Journal Item 2: with amount_currency: 10, currency_id: VND
Reconcile Item 1 and Item 2.

The reconciliation is not made because one of the two lines has a foreign currency (USD). It means the reconciliation will be made in USD so 100 USD versus 0 USD. Nothing to reconcile then. Even with your fix, this use case leads to the same result. Can you tell me what is the expected results you want and why plz?

Reply:
@ngochung207 The problem is we always reconcile using the foreign currency if we can:
If the company's currency is VND:
(1) USD + VDN => reconciliation in USD (at least one foreign currency)
(2) USD + USD => reconciliation in USD (same foreign currency)
(3) EUR + USD => reconciliation in VND (2 foreign currencies)

Your example is case (1) and you don't have any amount to reconcile anything.  Indeed, 10 VND = 0 USD with a 1:23000 rate. So I think your problem is the reconciliation is made on USD and not on VND with your example.
However, I can't change it without breaking others reconciliations.
I could suggest to try a reconciliation in company's currency in (1) if the amounts are too tiny to reconcile anything in foreign currency but I have to check and challenge that case with an accounting product owner.

Reply:
@ngochung207 I checked with them. I'll produce a fix for that (hopefully this week)